### ISSUE TITLE
entrypoint.sh messes up the output loglevel

### ISSUE THREAD

**Issue by lordslair** [joeferner/redis-commander#468]
Platform: `Kubernetes`
version: `rediscommander/redis-commander:latest`

The `entrypoint.sh` script when starting the redis-commander.js overwrites the different loglevels used in the main code
Everything (INFO, ERROR, WARN) gets out as LOGLEVEL.Unknown when parsed by log analysis tools (like ELK stack, or Loki)

Is there a way, at your knowledge, either to:
- Kinda forward through the entrypoint script the original loglevels 
- Mounting over the configfile `config/custom-environment-variables.json` with my own file, and start redis-commander with `command: ["/usr/bin/node ./bin/redis-commander --redis-host SERVER"]`
- Starting redis-commander by providing only args and no ENV and in the same idea, override the container entrypoint command

I tried the second option. It kinda work, but it's messy and could easily break if in future releases the configfile evolves
The third looks better, and seems more practical, but it fails as the code looks by default for a conf file, so I'm back to option two

Maybe there's an even better solution, or it's something you tried and for some reason I can't grasp it inherently won't work

If you need more details in case I wasn't descriptive enough, feel free to tell me

**Comment by sseide** [comment]
sorry, i do not really understand what you mean with `entrypoint.sh` overwrites something.

There are no environment variables to define a log level. Redis-Commander does not implement proper logging wih differen log levels right now (e.g. does not use winston or similar logging lib). It uses "console.log" and "console.error" only writing to STDOUT / STDERR respectively. There is no switch / config field or whateever to set any kind of log level as it does not know (right now) what this is.

**Comment by lordslair** [comment]
Sorry if I wasn't descriptive enough, I'll rephrase the issue

The actual redis-commander using console.{log,error} is not the issue, and is enough

The issue is: 
If you start **with entrypoint.sh** redis-commander: the bash script wraps the initial command, and send all messages to STDOUT only
But if you start **directly** redis-commander : `/usr/bin/node ./bin/redis-commander`, logs are sent to STDOUT / STDERR - as you described in your answer

What I try to achieve / Why I opened the issue:
Differentiate 'console.log > STDOUT' and 'console.error > STDERR' to be able to filter the logs when using log analysis tools

What I did:
I bypassed the `entrypoint.sh` and started directly the container using `/usr/bin/node ./bin/redis-commander`, and providing options --redis-port, --redis-host, and so on
Now I receive correctly the STDERR logs as they should, and separate the warnings for the rest as you can see in the screenshot below
![loki-redis-commander](https://user-images.githubusercontent.com/5469390/155180142-377f9dd2-048f-423c-b99c-315011a8332f.png)

TL;DR: if someone wants to receive the original STDOUT / STDERR, that's a quick workaround

**Comment by sseide** [comment]
ok, now i understand better. i will have a look into it.

**Comment by sseide** [comment]
cannot reproduce locally it seems. Tested with a normal docker setup locally.

To force some `console.err()` messages i modified the `local.json` config mounted into the container and changed a boolean value into a string (`"noSave": false` => `"noSave": "fals"`). This triggers an error in config file validation early on app startup.

I started the container with 
```sh
docker run -p 8081:8081 -v config/local.json:/redis-commander/config/local.json rediscommander/redis-commander
```

Now all logs of this container are written by docker with the json-file driver into its own logfile at `/var/lib/docker/containers/<imageid>/<imageid>-json.log`. 

Get the imageid with:
```sh
$ docker ps -a | grep redis-com | head -1 | rev | awk '{print $1}' | rev
$ docker inspect <container_name> | jq '.[].Id` 
```

opening the log i see some messages going to `stdout` and some to `stderr` as expected (the stream field denotes to source of this log message)...

```javascript
{"log":"node ./bin/redis-commander  \n","stream":"stdout","time":"2022-02-23T20:53:47.486362556Z"}
{"log":"Config key \"noSave\" has invalid value. Value must be boolean!\n","stream":"stderr","time":"2022-02-23T20:53:47.74987812Z"}
{"log":"Attention - insecure \"none\" algorithm allowed to check external SSO JWT token\n","stream":"stderr","time":"2022-02-23T20:53:4
7.750640525Z"}
{"log":"Configuration invalid - 1 errors found.\n","stream":"stderr","time":"2022-02-23T20:53:47.750815137Z"}
```

**Comment by lordslair** [comment]
You're right
I installed a Docker stack locally to reproduce your test scenario, and the Docker generated logs are sent where they belong - in `stdout` or in `stderr`

So it seems either it's something related to my k8s cluster logging config, or it's a difference between the docker logging system and the k8s one (as they deprecated docker and switched to containerd to manage the runtimes)

I'm sorry I bothered you with this issue which looks like not to be on your side
Thanks for the checks, I think you can close the issue

Kudos for this project btw
Long live redis-commander

**Comment by sseide** [comment]
Thanks and no problem. Seems to be a problem with the logging driver used than...
